File: frontend/demo/component/badge/badge-size.ts
import 'Frontend/demo/init'; // hidden-source-line
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('badge-size')
export class Example extends LitElement {
  protected createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  render() {
    return html`
      <!-- tag::snippet[] -->
      <span theme="badge small">Pending</span>
      <span theme="badge success small">Confirmed</span>
      <span theme="badge error small">Denied</span>
      <span theme="badge contrast small">On hold</span>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/badge/BadgeSize.java
package com.vaadin.demo.component.badge;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Span;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;

@Route("badge-size")
public class BadgeSize extends HorizontalLayout {

    public BadgeSize() {
        // tag::snippet[]
        Span pending = new Span("Pending");
        pending.getElement().getThemeList().add("badge small");

        Span confirmed = new Span("Confirmed");
        confirmed.getElement().getThemeList().add("badge success small");

        Span denied = new Span("Denied");
        denied.getElement().getThemeList().add("badge error small");

        Span onHold = new Span("On hold");
        onHold.getElement().getThemeList().add("badge contrast small");
        // end::snippet[]

        add(pending, confirmed, denied, onHold);
    }
    public static class Exporter extends DemoExporter<BadgeSize> { // hidden-source-line
    } // hidden-source-line
}
